src/random.jl

```diff
@@ -114,3 +114,69 @@ Return BitVector of primary producers.
 function whoisproducer(A)
     vec(.!any(A, dims=2))
 end
+
+"""
+**Create facilitation links**
+
+Takes trophic matrix (A) and number or connectance of facilitation links (resp. L and C).
+Returns matrix of facilitation links.
+Facilitating species are in rows, facilitated species are in columns.
+"""
+function facilitationlinks(A, L::Int64)
+
+    # Test.
+    @assert L >= 0
+
+    # Find candidate links.
+    S = numberspecies(A)
+    facilitation_matrix = zeros(Int, S, S) # initialize
+    candidates = wherecanfacilitate(A)
+    size(candidates, 1) < L ? throw(ArgumentError("L is too large.")) : nothing
+
+    # Choose randomly between candidates.
+    choosen = sample(candidates, L, replace=false)
+    for l in 1:L
+        i, j = choosen[l]
+        facilitation_matrix[i, j] = 1
+    end
+
+    facilitation_matrix
+end
+
+function facilitationlinks(A, C::Float64)
+
+    # Test. Connectance is in [0,1]?
+    @assert C >= 0.0
+    @assert C <= 1.0
+
+    # Create facilitation matrix.
+    S = numberspecies(A)
+    L = C * (S^2) # number of facilitation links
+    L = Int64(round(L)) # formatting
```

Comment:
You should rather directly round to desired type with [`round(Int64, L)`](https://docs.julialang.org/en/v1/base/math/#Base.round-Tuple{Type,%20Any}) ;)